zookeeper-server/src/test/java/org/apache/zookeeper/test/MultiTransactionTest.java

```diff
@@ -154,8 +154,17 @@ public void processResult(int rc, String path, Object ctx,
                 zk.multi(ops);
                 Assert.fail("Shouldn't have validated in ZooKeeper client!");
             } catch (KeeperException e) {
-                Assert.assertEquals("Wrong exception", expectedErr, e.code()
-                        .name());
+                String[] codeAndPath = expectedErr.split("@");
+                if (codeAndPath.length < 1 || codeAndPath.length > 2) {
+                    Assert.fail("Invalid exceptedErr=" + expectedErr);
+                }
+                Assert.assertEquals("Wrong exception", codeAndPath[0],
```

Comment:
You shouldn't check for string match if you can access the actual type. Please use `assertThat(..., instanceOf(...))` or `assertThat(..., isA(...))`.

Reply:
Addressed comment